iphone/cli/hooks/storyboard.js

```diff
@@ -0,0 +1,591 @@
+		const loaded = await this.loadStoryboardMetadata();
+		if (!loaded) {
+			return this.doFullTaskRun();
+		}
+
+		this._storyboards.forEach(storyboardInfo => {
+			if (!fs.existsSync(storyboardInfo.path)) {
+				this.logger.trace(`Storyboard at ${storyboardInfo.path} deleted, removing metadata`);
+				this._storyboards.delete(storyboardInfo.name);
+				return;
+			}
+
+			// Remove any storyboards from deactivated modules
+			if (!this._storyboardPaths.has(storyboardInfo.path)) {
+				this.logger.trace(`Storyboard at ${storyboardInfo.path} no longer in search path, removing metadata`);
+				this._storyboards.delete(storyboardInfo.name);
+				return;
+			}
+		});
+
+		const changedStoryboards = new Set();
+		changedFiles.forEach((fileStatus, pathAndFilename) => {
+			if (fileStatus === 'created' || fileStatus === 'changed') {
+				const packageExtension = '.storyboard';
+				const storyboardPath = pathAndFilename.substring(0, pathAndFilename.indexOf(packageExtension) + packageExtension.length);
+				if (!changedStoryboards.has(storyboardPath)) {
+					this.logger.trace(`Storyboard at ${storyboardPath} changed, regenerating metadata`);
+					changedStoryboards.add(storyboardPath);
+				}
+			}
+		});
+
+		await this.inspectStoryboards(changedStoryboards);
+		return this.writeStoryboardMetadata();
+	}
+
+	/**
+	 * @return {Promise<void>}
+	 * @inheritdoc
+	 */
+	async loadResultAndSkip() {
+		const loaded = await this.loadStoryboardMetadata();
+		if (!loaded) {
+			return this.doFullTaskRun();
+		}
+	}
+
+	/**
+	 * Loads stored metadata from disk and recreates the {@link StoryboardInfo}
+	 * objects.
+	 *
+	 * @return {Promise<Boolean>} True if the metadata was sucessfully loaded, false if not
+	 * @access private
+	 */
+	async loadStoryboardMetadata() {
+		try {
+			const metadata = await fs.readJSON(this._metadataPathAndFilename);
+			for (const storyboardPath of Object.keys(metadata)) {
+				const storyboardMetadata = metadata[storyboardPath];
+				this._storyboards.set(storyboardMetadata.name, new StoryboardInfo(
+					storyboardMetadata.name,
+					storyboardMetadata.path
+				));
+			}
+			return true;
+		} catch (e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Saves the internal matadata object to disk for reuse on subsequent builds.
+	 *
+	 * @return {Promise<void>}
+	 * @access private
+	 */
+	async writeStoryboardMetadata() {
+		const metadataObject = {};
+		for (const storyboardInfo of this._storyboards.values()) {
+			metadataObject[storyboardInfo.path] = {
+				name: storyboardInfo.name,
+				path: storyboardInfo.path
+			};
+		}
+		await fs.ensureDir(this._outputDirectory);
+		return fs.writeJSON(this._metadataPathAndFilename, metadataObject);
+	}
+
+	/**
+	 * Inspects each storyboard for their type and supported architectures.
+	 *
+	 * @param {Set.<String>} storyboardPaths - List of storyboard paths to inspect
+	 * @return {Promise}
+	 * @access private
+	 */
+	inspectStoryboards(storyboardPaths) {
```

Comment:
why not make `async` so you can `await` rather than use the `then` call in the loop?

Reply:
Done